core/trino-main/src/main/java/io/trino/operator/join/SpilledLookupSourceHandle.java

```diff
@@ -85,14 +87,25 @@ public synchronized void dispose()
     {
         disposeRequested.set(null);
         unspilledLookupSource = null; // let the memory go
-        setState(State.DISPOSED);
+        setState(State.DISPOSE_REQUESTED);
     }
 
     public SettableFuture<Void> getDisposeRequested()
     {
         return disposeRequested;
     }
 
+    public synchronized void setDisposeCompleted()
+    {
+        disposeCompleted.set(null);
+        setState(State.DISPOSE_COMPLETED);
```

Comment:
can this be called before `dispose` by `HashBuilderOperator`?

Reply:
why do you want to call this before `dispose`? Then the next page could be loaded before `dispose` is done.